src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/token.py

```diff
@@ -0,0 +1,310 @@
+    if password2:
+        new_passwords = [password for password in new_passwords if password.name != "password2"]
+
+    new_passwords_payload = []
+    for password in new_passwords:
+        new_passwords_payload.append({
+            "Name": password.name
+        })
+
+    token_update_parameters = {
+        "Credentials": {
+            "Certificates": new_certificates_payload,
+            "Passwords": new_passwords_payload
+        }
+    }
+
+    from msrest.exceptions import ValidationError
+    try:
+        return client.update(
+            resource_group_name,
+            registry_name,
+            token_name,
+            token_update_parameters
+        )
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_token_credential_add_certificate(cmd,
+                                         client,
+                                         registry_name,
+                                         token_name,
+                                         certificate1=None,
+                                         certificate2=None,
+                                         resource_group_name=None):
+
+    if certificate1 is None and certificate2 is None:
+        raise CLIError("At least one of the certificates must be provided")
+
+    token = acr_token_show(cmd,
+                           client,
+                           registry_name,
+                           token_name,
+                           resource_group_name)
+
+    certificates_dict = {}
+    for certificate in token.credentials.certificates:
+        certificates_dict[certificate.name] = None
+    certificates_dict = _handle_add_certificate(certificates_dict, certificate1, certificate2)
+
+    certificates_payload = []
+    for key in certificates_dict:
+        value = certificates_dict[key]
+        if value:
+            certificate = {
+                "Name": key,
+                "EncodedPEMCertificate": value
+            }
+        else:
+            certificate = {
+                "Name": key
+            }
+        certificates_payload.append(certificate)
+
+    certificates_payload.sort(key = lambda cert: cert["Name"])
+
+    token_update_parameters = {
+        "Credentials": {
+            "Certificates": certificates_payload
+        }
+    }
+    
+    from msrest.exceptions import ValidationError
+    try:
+        return client.update(
+            resource_group_name,
+            registry_name,
+            token_name,
+            token_update_parameters
+        )
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def _handle_add_certificate(certificates_dict, certificate1, certificate2):
+    from base64 import b64encode
+
+    if certificate1:
+        try:
+            certificate1_content = open(certificate1, "r").read()
+            certificate1_b64 = b64encode(certificate1_content.encode())
+            certificate1_b64 = str(certificate1_b64.decode())
```

Comment:
[bytes.decode](https://docs.python.org/3/library/stdtypes.html#bytes.decode) gives a `str` back anyway

Reply:
Oh, awesome. Fixed. Thank you!